src/domain/authorizations/authorizations.js

```diff
@@ -0,0 +1,125 @@
+const util = require('util')
+
+const Config = require('../../lib/config.js')
+const { getStackOrInspect } = require('../../lib/util')
+
+/**
+ * Forwards GET authorizations endpoint requests to destination FSP for processing
+ *
+ * @returns {undefined}
+ */
+const forwardAuthorizationMessage = async (headers, transactionRequestId, payload, method) => {
+  let endpoint
+  const fspiopSource = headers[Enum.Http.Headers.FSPIOP.SOURCE]
+  const fspiopDest = headers[Enum.Http.Headers.FSPIOP.DESTINATION]
+  const messageType = method === Enum.Http.RestMethods.GET ? 'request' : 'response'
+  const payloadLocal = method === Enum.Http.RestMethods.GET ? undefined : payload
+
+  try {
+    endpoint = await Endpoint.getEndpoint(Config.SWITCH_ENDPOINT, fspiopDest, Enum.EndPoints.FspEndpointTypes.FSPIOP_CALLBACK_URL_AUTHORIZATIONS)
+    Logger.info(`Resolved party ${Enum.EndPoints.FspEndpointTypes.FSPIOP_CALLBACK_URL_AUTHORIZATIONS} endpoint for authorizations ${messageType} ${transactionRequestId || 'error.test.js'} to: ${util.inspect(endpoint)}`)
+    if (!endpoint) {
+      throw ErrorHandler.Factory.createFSPIOPError(ErrorHandler.Enums.FSPIOPErrorCodes.DESTINATION_FSP_ERROR, `No ${Enum.EndPoints.FspEndpointTypes.FSPIOP_CALLBACK_URL_AUTHORIZATIONS} endpoint found for transactionRequest ${transactionRequestId} for ${Enum.Http.Headers.FSPIOP.DESTINATION}`, undefined, fspiopSource)
+    }
+    const query = method === Enum.Http.RestMethods.GET ? '?' + (new URLSearchParams(payload).toString()) : ''
+    const fullUrl = `${endpoint}/authorizations/${transactionRequestId}${query}`
+    Logger.info(`Forwarding authorization request to endpoint: ${fullUrl}`)
+    // Network errors lob an exception. Bare in mind 3xx 4xx and 5xx are not network errors
+    // so we need to wrap the request below in a `try catch` to handle network errors
+    let res
+    try {
+      res = await requests.sendRequest(fullUrl, headers, fspiopSource, fspiopDest, method, payloadLocal)
+    } catch (e) {
+      throw ErrorHandler.Factory.createFSPIOPError(ErrorHandler.Enums.FSPIOPErrorCodes.DESTINATION_COMMUNICATION_ERROR, `Network error forwarding authorization ${messageType}: ${getStackOrInspect(e)}`, 'Network error', fspiopSource)
+    }
+    Logger.info(`Forwarding authorization ${messageType} for transactionRequestId ${transactionRequestId} from ${fspiopSource} to ${fspiopDest} got response ${res.status} ${res.statusText}`)
+    // handle non network related errors below
+    if (!res.ok) {
+      throw ErrorHandler.Factory.createFSPIOPError(ErrorHandler.Enums.FSPIOPErrorCodes.DESTINATION_COMMUNICATION_ERROR, 'Got non-success response forwarding authorization ' + messageType, res.statusText, fspiopSource)
+    }
+    return true
+  } catch (err) {
+    Logger.info(`Error forwarding authorization ${messageType} to endpoint ${endpoint}: ${getStackOrInspect(err)}`)
+    const errorHeaders = Object.assign({}, headers, {
```

Comment:
This can be switched out to more readable spread notation:
```js
const errorHeaders = {
  ...headers,
   'fspiop-source': Enum.Http.Headers.FSPIOP.SWITCH.value,
   'fspiop-destination': fspiopSource
}
```